core/modules/replica/DatabaseServicesMySQL.cc

```diff
@@ -2243,57 +2243,64 @@ vector<TransactionContribInfo> DatabaseServicesMySQL::transactionContribs(
 }
 
 
-TransactionContribInfo DatabaseServicesMySQL::beginTransactionContrib(
-        TransactionId transactionId,
-        string const& table,
-        unsigned int chunk,
-        bool isOverlap,
-        string const& worker,
-        string const& url) {
+TransactionContribInfo DatabaseServicesMySQL::createdTransactionContrib(
+        TransactionContribInfo const& info,
+        bool failed) {
```

Comment:
I think it would be better to modify ` TransactionContribInfo& info` instead of returning TransactionContribInfo, but it doesn't matter much either way.

Reply:
Unfortunately, method `DatabaseServicesMySQL::createdTransactionContrib` can't return the `non-const` reference to the input object (of class `TransactionContribInfo`) that is passed into the method via the `const` reference. And it's probably better to keep the code this way by separating the non-mutable input from the output returned by value. Though, I don't have a strong preference for that.